File: go/two-fer/two_fer.go
// Package twofer provides a simple implementation of the
// common form of "two for one"
package twofer

// ShareWith puts together a sentence with x as placeholder.
// If no argument is given, x defaults to "you".
// Returns following string "One for x, one for me."
func ShareWith(name string) string {
	if name == "" {
		name = "you"
	}

	return "One for " + name + ", one for me."
}

File: php/two-fer/two-fer.php
<?php

/**
 * Given a name, return a string with the message: One for X, one for me.
 *
 * @param string $name The string to create the message with.
 *
 * @return string The result as string.
 */
function twoFer(string $name = "you"): string {
  return "One for ${name}, one for me.";
}
